js/ui/status/keyboard.js

```diff
@@ -527,6 +576,14 @@ const InputSourceManager = new Lang.Class({
 
     _inputSourcesChanged: function() {
         let sources = this._settings.inputSources;
+        if (this._needsFallbackSource) {
```

Comment:
I think it would be possible and cleaner to do this together with the `infosList.length == 0` case below. It looks like the same input source would be created in that case.

Reply:
Good catch, I merged the passwordFallbackCreation code into that one. I store the source information into this._fallbackInputSource so we can activate it from _syncPasswordMode. I was wondering if I should store just the index but then I would need to access this._inputSources directly outside of inputŚourcesChanged.